src/Pyz/Yves/ShopUi/Theme/default/components/molecules/date-time-picker/date-time-picker.ts

```diff
@@ -1,115 +1,71 @@
 import Component from 'ShopUi/models/component';
-import $ from 'jquery/dist/jquery';
-import 'jquery-datetimepicker/build/jquery.datetimepicker.full';
 
 export default class DateTimePicker extends Component {
     protected trigger: HTMLInputElement;
     protected dateFrom: HTMLInputElement;
     protected dateTo: HTMLInputElement;
+    protected datePicker: HTMLInputElement;
 
     protected readyCallback(): void {}
 
     protected init(): void {
-        this.trigger = <HTMLInputElement>this.getElementsByTagName('input')[0];
-        this.dateFrom = <HTMLInputElement>document.getElementById(this.dateFromId);
-        this.dateTo = <HTMLInputElement>document.getElementById(this.dateToId);
+        this.trigger = this.querySelector<HTMLInputElement>('input[type="text"]');
+        this.dateFrom = document.getElementById(this.dateFromId) as HTMLInputElement;
+        this.dateTo = document.getElementById(this.dateToId) as HTMLInputElement;
+        this.datePicker = this.querySelector<HTMLInputElement>(`.${this.jsName}__datepicker`);
+        this.datePicker.value = this.trigger.value;
 
+        this.datePickerInit();
         this.mapEvents();
     }
 
     protected mapEvents(): void {
-        this.datetimepickerInit();
-        this.setLanguage(this.language);
-
         if (this.dateTo) {
             this.setMaxDate();
+            this.dateTo.addEventListener('change', () => this.setMaxDate());
         }
         if (this.dateFrom) {
             this.setMinDate();
+            this.dateFrom.addEventListener('change', () => this.setMinDate());
         }
-    }
 
-    protected datetimepickerInit(): void {
-        if (this.formattedDateTime && $(this.trigger).val()) {
-            $(this.trigger).val(this.formattedDateTime);
-        }
+        this.trigger.addEventListener('focus', () => {
+            this.datePicker.showPicker();
+        });
 
-        $(this.trigger).datetimepicker(this.config);
+        this.trigger.addEventListener('click', () => {
+            this.datePicker.showPicker();
+        });
+
+        this.datePicker.addEventListener('change', () => {
+            this.trigger.value = this.datePicker.value;
+        });
+
+        this.trigger.addEventListener('change', () => {
+            this.datePicker.value = this.trigger.value;
+        });
     }
 
-    protected setLanguage(language: string): void {
-        $.datetimepicker.setLocale(language);
+    protected datePickerInit(): void {
```

Comment:
Do we need this method, as looks like some part of initializing you are doing  in the `mapEvents` (setMinDate, setMaxDate), some oitside

Reply:
Moved initializing for properties into the method.